cfme/tests/automate/custom_button/test_import_export.py

```diff
@@ -0,0 +1,122 @@
+import pytest
+import fauxfactory
+
+from widgetastic_patternfly import Dropdown
+
+from cfme.infrastructure.provider.virtualcenter import VMwareProvider
+from cfme.markers.env_markers.provider import ONE_PER_TYPE
+from cfme.utils.appliance.implementations.ui import navigate_to
+
+
+pytestmark = [
+    pytest.mark.tier(2),
+    pytest.mark.usefixtures("setup_provider"),
+    pytest.mark.provider([VMwareProvider], selector=ONE_PER_TYPE),
+]
+
+
+@pytest.fixture(scope="function")
+def setup_groups_buttons(appliance, provider):
+    collection = appliance.collections.button_groups
+    gp_buttons = {}
+
+    for obj_type in ["PROVIDER", "VM_INSTANCE"]:
+        gp = collection.create(
+            text=fauxfactory.gen_alphanumeric(),
+            hover=fauxfactory.gen_alphanumeric(),
+            type=getattr(collection, obj_type),
+        )
+
+        button = gp.buttons.create(
+            text=fauxfactory.gen_alphanumeric(),
+            hover=fauxfactory.gen_alphanumeric(),
+            display_for="Single and list",
+            system="Request",
+            request="InspectMe",
+        )
+        if obj_type == "PROVIDER":
+            obj = provider
+        else:
+            obj = appliance.provider_based_collection(provider).all()[0]
```

Comment:
This could cause an IndexError during setup if there are no VM's listed by all() method.

Please wrap in a `try` block, and call pytest.skip() in that instance, to cleanly skip the test when the provider doesn't have the required resources available.

Slim chance, but this is something I would like to start enforcing more strictly across our test fixtures.